next-app/pages/index.js

```diff
@@ -4,17 +4,33 @@ import { useContext, useEffect } from 'react';
 import Nav from '@/components/Nav';
 import { GeneralDataContext } from '@/components/WordDataProvider';
 
+import { Messenger } from '@/lib/messenger';
+
 export default function IndexPage () {
   const {activePage, setActivePage} = useContext(GeneralDataContext);
 
+  if (typeof window !== 'undefined') {
+    const messenger = new Messenger("index", "background");
+    messenger.listen();
+    messenger.registerEvent(["get-stored-data"]);
+    messenger.addEventListener("get-stored-data", (message) => {
+      console.log("index received stored data");
+      if (!message.data["keyPairs"]) {
+        console.log("no keypairs");
+        setActivePage("welcome");
+      } else {
+          let apiKey = message.data["keyPairs"][0]["apiKey"];
+          let host = message.data["keyPairs"][0]["hostKey"];
```

Comment:
we currently set keyPairs but also an explicit apiKey and hostKey. Because we will have multiple pairs, the active one will be stored into those separate apiKey and hostKey. We should take data from there, so if we have multiple pairs and change the active one we actually use the updated version of credentials.

Reply:
i think we should switch to an object in this case instead of an array (?) because then we have a structure of keyPairs = {api: "en", key: "...", host: "..."}, {api: "de", key: "...", host: "..."}}